_extensions/diagram/diagram.lua

```diff
@@ -152,17 +152,44 @@ local mermaid = {
     local mime_type = self.mime_type or 'image/svg+xml'
     local file_extension = extension_for_mimetype[mime_type]
     return with_temporary_directory("diagram", function (tmpdir)
-      return with_working_directory(tmpdir, function ()
-        local infile = 'diagram.mmd'
-        local outfile = 'diagram.' .. file_extension
-        write_file(infile, code)
-        pipe(
-          self.execpath or 'mmdc',
-          {"--pdfFit", "--input", infile, "--output", outfile},
-          ''
-        )
-        return read_file(outfile), mime_type
-      end)
+      local infile = tmpdir .. '/diagram.mmd'
+      local outfile = tmpdir .. '/diagram.' .. file_extension
+	  --- Configure options for mmdc based on engine options
+      local args = {'--pdfFit', '--input', infile, '--output', outfile}
+	  setmetatable(args, {__index = {insert = table.insert}})
+      if self.opt then
+        if self.opt['theme'] then
```

Comment:
The repeated `if` statements seems a bit clunky to me. How about defining a table that maps option keys to command line arguments, and then loop over that table in a for loop? Something along the lines of

``` lua
local options = { theme = '--theme', ...}
for opt, cliopt in pairs(options) do
  ...
end
```